pyocd/debug/semihost.py

```diff
@@ -184,6 +185,9 @@ def open(self, fnptr, fnlen, mode):
             return fd
 
         try:
+            # ensure directories are exists
+            pathlib.Path(filename).parent.mkdir(parents=True, exist_ok=True)
```

Comment:
This should probably only be done if `mode` is writable, eg `('w' in mode) or ('a' in mode)`.

Reply:
thanks for your comments.
updated.